src/state/multicall/hooks.ts

```diff
@@ -162,21 +163,31 @@ export function useSingleContractMultipleData(
 ): CallState[] {
   const fragment = useMemo(() => contract?.interface?.getFunction(methodName), [contract, methodName])
 
-  const blocksPerFetch = options?.blocksPerFetch
+  // encode callDatas
+  const callDatas = useMemo(
+    () =>
+      callInputs.every((callInput) => isValidMethodArgs(callInput)) && fragment
```

Comment:
idr why we check that every input is a valid list of args instead of just encoding the ones that are valid and ignoring the ones that aren't

Reply:
yeah in theory i think we could do that and use the `Call | undefined` return sig like in `useMultipleContractSingleData`, not sure it's worth it though

Reply:
ok addressed actually